pkg/provider/cache/cache.go

```diff
@@ -0,0 +1,99 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
```

Comment:
This PR was done before generators IIRC - having said that, I believe we do want to use the same caching mechanism for the generators (so I'm not sure we want to have feature/cache under provider) :)

Reply:
I moved it one level up into `./pkg`; In fact, it is already used by the ecr generator.